src/Storages/ObjectStorageQueue/ObjectStorageQueueOrderedFileMetadata.cpp

```diff
@@ -604,19 +742,38 @@ void ObjectStorageQueueOrderedFileMetadata::filterOutProcessedAndFailed(
     {
         const auto & path = paths[i];
         const auto bucket = use_buckets_for_processing ? getBucketForPathImpl(path, buckets_num) : 0;
-        if (!max_processed_file_per_bucket.empty()
-            && path <= max_processed_file_per_bucket[bucket])
+        if (!max_processed_file_per_bucket_and_hive_partition.empty())
         {
-            LOG_TEST(log_, "Skipping file {}: Processed", path);
-            continue;
+            if (is_path_with_hive_partitioning)
+            {
+                std::string hive_part(getHivePart(path));
+                auto max_processed_file = max_processed_file_per_bucket_and_hive_partition[bucket].find(std::string(hive_part));
+                if (max_processed_file != max_processed_file_per_bucket_and_hive_partition[bucket].end()
+                    && path < max_processed_file->second)
```

Comment:
Shouldn't it be like this?
```suggestion
                    && path <= max_processed_file->second)
```